eth/api.go

```diff
@@ -694,6 +695,12 @@ func (s *PublicBlockChainAPI) doCall(args CallArgs, blockNr rpc.BlockNumber) (st
 	}
 	stateDb = stateDb.Copy()
 
+	// If there's no code to interact with, respond with an appropriate error
+	if args.To != nil {
+		if code := stateDb.GetCode(*args.To); len(code) == 0 {
+			return "0x", nil, bind.ErrNonExistentContract
```

Comment:
Hmm, perhaps it's still possible since eth wouldn't depend on the backends package, only on bind. But the tests would still need to go then into eth_test to be able to pull in the simulator (or we don't test the contract integration :P)